core/src/main/java/io/confluent/rest/ApplicationGroup.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.rest;
+
+import org.eclipse.jetty.server.Server;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class ApplicationGroup<T extends RestConfig> {
+  private final T config;
+  private Server server = null;
+  private final List<Application> applications = new ArrayList<>();
+
+  ApplicationGroup(T config, Application ...applications) {
+    this.config = config;
+    this.applications.addAll(Arrays.asList(applications));
+    System.out.println(applications.toString());
+  }
+
+  public void addApplication(Application application) {
+    this.applications.add(application);
+  }
+
+  public List<Application> getApplications() {
+    return applications;
+  }
+
+  public RestConfig getConfiguration() {
```

Comment:
I'm confused to why an group has a configuration. I was thinking that only `Application` and `ApplicationServer` have configurations, for app specific settings and global settings respectively. `ApplicationGroup` itself is not doing anything with the configuration, so I guess it does not belong here.

Reply:
There is nothing preventing App1 and App2 from having differing configurations with regard to how the shared resource, the ApplicationServer itself, is configured. Rather than picking adding some logic to pick a winner I make the developer explicitly decide which configuration will be used to setup the ApplicaitonServer.